ui/app/components/app/modals/customize-nonce/customize-nonce.component.js

```diff
@@ -0,0 +1,159 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import Modal from '../../modal';
+import TextField from '../../../ui/text-field';
+import Button from '../../../ui/button';
+import Typography from '../../../ui/typography';
+import {
+  COLORS,
+  TYPOGRAPHY,
+  FONT_WEIGHT,
+  ALIGN_ITEMS,
+  BLOCK_SIZES,
+} from '../../../../helpers/constants/design-system';
+import Box from '../../../ui/box';
+
+export default class CustomizeNonce extends PureComponent {
+  static propTypes = {
+    hideModal: PropTypes.func.isRequired,
+    customNonceValue: PropTypes.string,
+    nextNonce: PropTypes.number,
+    updateCustomNonce: PropTypes.func,
+    getNextNonce: PropTypes.func,
+  };
+
+  static contextTypes = {
+    t: PropTypes.func,
+  };
+
+  state = {
+    customNonceValue: '',
+  };
+
+  onChange = (e) => {
+    this.setState({
+      customNonceValue: e.target.value,
+    });
+  };
+
+  onSubmit = () => {
+    const { customNonceValue } = this.props;
+    if (this.state.customNonceValue === '') {
+      this.props.updateCustomNonce(customNonceValue);
+    } else {
+      this.props.updateCustomNonce(this.state.customNonceValue);
+    }
+    this.props.getNextNonce();
+    this.props.hideModal();
+  };
+
+  onReset = () => {
+    const { nextNonce } = this.props;
+    this.setState({
+      customNonceValue: nextNonce,
+    });
+    document.getElementById('custom-nonce-id').value = nextNonce;
```

Comment:
as discussed on a call, it's interesting that this line is needed at all, given `this.state.customNonceValue` is tied the `value` prop of that `TextField`